rust/metering/tests/tests.rs

```diff
@@ -0,0 +1,298 @@
+// NOTE(c-gamble): Procedural macros cannot be used in the same crates in which they are defined.
+// Instead, it is recommended to create a `tests/` directory in which to write unit and integration
+// tests. See https://github.com/rust-lang/rust/issues/110247 for additional information.
+#[cfg(test)]
+use async_trait::async_trait;
+use chroma_metering::initialize_metering;
+use std::sync::{Arc, Mutex};
+use tracing::Span;
+
+use crate::metering::{MeteredFutureExt, SubmitExt};
+/// Represents a user defining their own metering module.
+mod metering {
+    use super::initialize_metering;
+
+    initialize_metering! {
+        #[attribute(name = "test_attribute")]
+        type TestAttribute = Option<String>;
+
+        #[event]
+        #[derive(Default, Debug, Clone)]
+        pub struct TestEventA {
+            // This field is not read in the tests
+            #[allow(dead_code)]
+            test_unannotated_field: u64,
+            #[field(attribute = "test_attribute", mutator = "test_mutator_a")]
+            pub test_annotated_field: Option<String>
+        }
+
+        #[event]
+        #[derive(Default, Debug, Clone)]
+        pub struct TestEventB {
+            // This field is not read in the tests
+            #[allow(dead_code)]
+            test_unannotated_field: u64,
+            #[field(attribute = "test_attribute", mutator = "test_mutator_b")]
+            pub test_annotated_field: Option<String>
+        }
+    }
+
+    fn test_mutator_a(event: &mut TestEventA, value: Option<String>) {
+        event.test_annotated_field = value;
+    }
+
+    fn test_mutator_b(event: &mut TestEventB, value: Option<String>) {
+        event.test_annotated_field = value;
+    }
+}
+
+#[derive(Clone, Debug)]
+struct TestReceiver {
+    messages: Arc<Mutex<Vec<String>>>,
+}
+
+impl TestReceiver {
+    pub fn new() -> Self {
+        Self {
+            messages: Arc::new(Mutex::new(Vec::new())),
```

Comment:
If the intent is to use this as a `Component` in `System` you don't need a Mutex here.